src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -217,73 +204,104 @@ public virtual DatabaseModel Create(DbConnection dbConnection, IEnumerable<strin
 
         #region Type information queries
 
-        IEnumerable<DatabaseTable> GetTables(
-            NpgsqlConnection connection,
-            Func<string, string, string> tableFilter)
+        /// <summary>
+        /// Queries the database for defined tables and registers them with the model.
+        /// </summary>
+        /// <param name="connection">The database connection.</param>
+        /// <param name="tableFilter">The table filter fragment.</param>
+        /// <param name="enums">The collection of discovered enums.</param>
+        /// <param name="logger">The diagnostic logger.</param>
+        /// <returns>
+        /// A collection of tables defined in the database.
+        /// </returns>
+        [NotNull]
+        static IEnumerable<DatabaseTable> GetTables(
+            [NotNull] NpgsqlConnection connection,
+            [CanBeNull] Func<string, string, string> tableFilter,
+            [NotNull] [ItemNotNull] HashSet<string> enums,
+            [NotNull] IDiagnosticsLogger<DbLoggerCategory.Scaffolding> logger)
         {
-            var commandText = @"
-SELECT nspname, relname, description
+            var filter = tableFilter != null ? $"AND {tableFilter("ns.nspname", "cls.relname")}" : null;
+            var commandText = $@"
+SELECT
```

Comment:
I still agree with @YohDeadfall that these would be better on a single line, but it's not very important.